scripts/common_test.py

```diff
@@ -874,18 +875,50 @@ class ManagedProcessTests(test_utils.TestBase):
     POPEN_CALL = (
         collections.namedtuple('POPEN_CALL', ['program_args', 'kwargs']))
 
+    def assert_proc_was_managed_as_expected(
+            self, logs, pid,
+            manager_should_have_sent_terminate_signal=True,
+            manager_should_have_sent_kill_signal=False):
+        """Asserts that the process ended as expected.
+
+        Args:
+            logs: list(str). The logs emitted during the process's lifetime.
+            pid: int. The process ID to inspect.
+            manager_should_have_sent_terminate_signal: bool. Whether the manager
+                should have sent a terminate signal to the process.
+            manager_should_have_sent_kill_signal: bool. Whether the manager
+                should have sent a kill signal to the process.
+        """
+        proc_pattern = r'Process\((name=\'python\', )?pid=%d\)' % (pid,)
+
+        expected_patterns = []
+        if manager_should_have_sent_terminate_signal:
+            expected_patterns.append(r'Terminating %s\.\.\.' % proc_pattern)
+        if manager_should_have_sent_kill_signal:
+            expected_patterns.append(r'Forced to kill %s!' % proc_pattern)
+        else:
+            expected_patterns.append('%s has ended\\.' % proc_pattern)
+
+        logs_with_pid = [msg for msg in logs if re.search(proc_pattern, msg)]
+        if expected_patterns and not logs_with_pid:
+            self.fail(msg='%r has no match in logs=%r' % (proc_pattern, logs))
+
+        self.assert_matches_regexps(expected_patterns, logs_with_pid)
```

Comment:
Also I don't really think it's worth making a new assert method in test utils just for this. Why not put the for loop here and have a bunch of individual asserts in the loop instead?

I think it's OK to have an assert_regexp_matches in the base test utils class, but I don't think you need to bring the for loop into it. The logic in the base class should be simple.

Reply:
So I see this pattern (match list of strings to list of regexps) often in the code base, so I think this is a very valuable assertion.

Adding individual string assertions one-by-one quickly became hard to manage while writing this test (and others in the past), so I'm confident it's worth keeping around and valuable to test writers.

Reply:
Ah, I see, OK.

Where else did you see it? If this is a common thing then it would be good to reuse it in a few more of those places (you don't need to find them all though).

Reply:
Yup, so the majority of our jobs use this pattern in tests, specifically when testing expected outputs. Those are going to be rewritten soon anyway by #11475, so when the time comes it'll already be valuable to them.

Reply:
To be honest, I think this is more of an API issue. I'd prefer if we could write tests closer to how we write Jasmine specs (expect(a).toBeLike(b)), but that'd require a third-party module like [PyHamcrest](https://github.com/hamcrest/PyHamcrest).

Reply:
Ah ok, that sounds reasonable (re the #11475 comment). LGTM then!

Prefer not to add more third-party libs though.

Reply:
Ack, that's fine with me too.